Source/Controllers/SPAppController.m

```diff
@@ -446,11 +442,12 @@ - (void)saveConnectionsToSPF:(NSNotification *)notification {
     [info setObject:@1 forKey:SPFVersionKey];
     [info setObject:@"connection bundle" forKey:SPFFormatKey];
 
-    // Loop through all windows
-    for (SPWindowController *windowController in [self.tabManager windowControllers]) {
+    NSMutableArray *tabs = [NSMutableArray array];
+    NSMutableDictionary *win = [NSMutableDictionary dictionary];
 
-        // First window is always the currently key window
-        NSMutableDictionary *win = [NSMutableDictionary dictionary];
+    // Loop through all windows / tabs
+    // TODO: Handle window vs. tab - for now we store all windows into one as tabs
```

Comment:
If I understand correctly, this really doesn't fully close the restoring issue then? Because now instead of always restoring as windows they'll just always be restored as tabs in a single window?

Reply:
I have a WIP to fix this all, but no, currently the restore doesn't work at all - it opens windows, but doesn't connect or anything like that.

Reply:
Fixed :)